### PR TITLE
 Die, FixedDimension, die, die, die!

### PR DESCRIPTION
**Pull request by mlange05**
I hope the title is self-explanatory. Should address issue #372.

### CODE REVIEW

**Comment by FabioLuporini** [comment]
Yes, there's no reason to have them and it's now just a maintenance burden  .

File: tests/test_dimension.py [hunk 1]
```diff
@@ -1,21 +0,0 @@
...
-    devito.x = FixedDimension(name='x', size=10, spacing=devito.x.spacing)
-    assert(devito.x in m.indices)
-
-
-@skipif_yask
-def test_correct_usage():
-    myx = FixedDimension(name='x', size=10, spacing=devito.x.spacing)
-    grid = Grid(shape=(10, 10), dimensions=(myx, y))
-    m = Function(name="m", grid=grid)
-    assert(myx in m.indices)
```
Comment (mloubout) [inline tests/test_dimension.py]: Maybe not for this PR, but should we still have at least one test (no idea what) for Dimensions?

Comment (mlange05) [reply tests/test_dimension.py]: We have a whole bunch of tests that use explicitly specified and grid-derived dimensions all over the test base - this test was really just specific to the fixed dimension API and it's recent change. I do agree though that we are lacking tests for some more exotic dimension combinations, in particular with respect to mixing space/time and auxiliary dimensions - but yes that is definitely stuff for future PRs.

Comment (mloubout) [reply tests/test_dimension.py]: > do agree though that we are lacking tests for some more exotic dimension combinations,

We made some with @fablup previously as part of some dse fixes. I think there is some left in #340 that include mixed dimensions laplacian. I can dig it up, and can be part of a new PR

Comment (mlange05) [reply tests/test_dimension.py]: Excellent. I'm still working on adding dedicated `TimeDimension`  types, so these will come in handy.

Comment (mloubout) [reply tests/test_dimension.py]: #340 is actually quite small, the `test_operator` with `b.laplace` would the go to one for this PR to make sure we catch correct dimensions for auto-shortcuts

**Comment by FabioLuporini** [comment]
> I do agree though that we are lacking tests for some more exotic dimension combinations, in particular with respect to mixing space/time and auxiliary dimensions

There are a few tests in ``test_operator.TestScheduler`` checking that code generation is appropriate with different types of dimensions, different order, etc. I agree this is far from being a complete coverage, though. And they're also not tailored to the new Dimension types...

**Comment by mlange05** [comment]
Ok, I take it everyone is happy with this then? @navjotk?

**Comment by navjotk** [comment]
Wait for 345 to go in first.

**Comment by mlange05** [comment]
Ok, with final approval this is good to go in, I think. Merging once travis is happy...

**Comment by mlange05** [comment]
Ok, merging...

**Pull request merged**